### PR TITLE
Upgrade CosmosDB Gremlin and Sql Container API version from `2021-10-15` to `2023-04-15` and use `go-azure-sdk`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
I intend to support new properties in latest API version. So I have to submit this PR to upgrade API version first.

image

### CODE REVIEW

File: internal/services/cosmos/common/autoscale_settings.go [hunk 1]
```diff
@@ -25,6 +26,23 @@ func ExpandCosmosDbAutoscaleSettings(d *pluginsdk.ResourceData) *documentdb.Auto
 	return &autoscaleSettings
 }
 
+func ExpandCosmosDbAutoscaleSettingsForGremlinAndSqlContainer(d *pluginsdk.ResourceData) *cosmosdb.AutoScaleSettings {
```
Comment (katbyte) [inline internal/services/cosmos/common/autoscale_settings.go:29]: This isn't for gremlin and SQL only? its for the new SDK? so can we keep the `ExpandCosmosDbAutoscaleSettingsResource` name and then rename the old one to `ExpandCosmosDbAutoscaleSettingsResourceLegacy` or `ExpandCosmosDbAutoscaleSettingsResourceTrack1` ?

this applies to all the new flatten and expand functions

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @neil-yechenwei - overall looks good with 1 comment about the new expand/flatten functions, once thats fixed this should be good to go!

File: internal/services/cosmos/common/autoscale_settings.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/cosmos/common/autoscale_settings.go]: Sure. Updated.

**Review by katbyte — APPROVED** [review]
Thanks ! LGTM 🔭

**Pull request merged**